cmd/monaco/cmdutils/cmdutils.go

```diff
@@ -32,30 +34,60 @@ func SilenceUsageCommand() func(cmd *cobra.Command, args []string) {
 	}
 }
 
-// VerifyClusterGen takes a manifestEnvironments map and tries to call the version endpoint of each environment
-// in order to verify that the user has configured the environments correctly.
-// Depending on the configured environment "type" the function tries to call the version endpoint of either
-// classic gen or platform gen. The function will return an error as soon as
-// it receives an error from calling the version endpoint of an environment
-func VerifyClusterGen(envs manifest.Environments) error {
+// VerifyEnvironmentGeneration takes a manifestEnvironments map and tries to verify that each environment can be reached
+// using the configured credentials
+func VerifyEnvironmentGeneration(envs manifest.Environments) bool {
```

Comment:
more conceptual question/thinking

Why print error message inside. I would go with returning error, than caller can decide what to do with error (e.g. which error lever to use for printing...)

Reply:
> more conceptual question/thinking
> 
> Why print error message inside. I would go with returning error, than caller can decide what to do with error (e.g. which error lever to use for printing...)

in general i agree with that, you need to do a bit of refactoring, and give the caller more information what went wrong, so that the caller is able to log stuff, tho